packages/wells/src/client/api/wells.ts

```diff
@@ -0,0 +1,178 @@
+          type: 'Well',
+        },
+        ...exactSearch,
+      },
+      search: {
+        ...fuzzySearch,
+      },
+    };
+    return await this.search(body);
+  };
+
+  /**
+   * Get wells filtering and/or with fuzzy search on name
+   *
+   * ```js
+   * const created = await client.wells.getWellByName('name');
+   * ```
+   *
+   * @param wellName the full name of the well
+   */
+  public getWellByName = async (wellName: string): Promise<Well[]> => {
+    const exactSearch = { name: wellName };
+    return Wells.mapToWell(await this.searchForWell(exactSearch));
+  };
+
+  /**
+   * Get a list of wells that contains the specified search prefix in the name
+   *
+   * ```js
+   * const created = await client.wells.getWellsByNamePrefix(namePrefix: 'somePrefix');
+   * ```
+   *
+   * @param namePrefix specify a prefix that the wellname should contain
+   */
+  public getWellsByNamePrefix = async (namePrefix: string): Promise<Well[]> => {
+    const fuzzySearch = { name: namePrefix };
+    const fuzzyResults = Wells.mapToWell(
+      await this.searchForWell({}, fuzzySearch)
+    );
+    return fuzzyResults.filter(function(well) {
+      return well.name.startsWith(namePrefix);
+    });
+  };
+
+  /**
+   * Get a multiple wells filtering and/or with fuzzy search on name
+   *
+   * ```js
+   * const created = await client.wells.getWellsByIds([{id: 123}, {externalId: 'abc'}]);
+   * ```
+   *
+   * @param ids contains unions of internal ids and external ids
+   */
+  public getWellsByIds = async (ids: IdEither[]): Promise<Well[]> => {
+    return Wells.mapToWell(await this.retrieve(ids));
+  };
+
+  /**
+   * Get a single well filtering and/or with fuzzy search on name
+   *
+   * ```js
+   * const created = await client.wells.getWellById(id: 123);
+   * ```
+   *
+   * @param id specific internal id for a particular well
+   */
+  public getWellById = async (id: number): Promise<Well[]> => {
+    return await this.getWellsByIds([{ id: id }]);
+  };
+
+  /**
+   * Receives a geoJson polygon like in Discover and return a list of well objects
+   *
+   * ```js
+   * const created = await client.wells.getWellsByPolygon(....);
+   * ```
+   *
+   * @param polygon lat and lon that make up a polygon
+   * @param source the source of spatial objects
+   * @param crs coordinate reference system of the spatial objects
+   * @param layerName the layer to which objects belong
+   * @param limit max number of objects to be returned
+   * @param offset the starting offset of objects in the results
+   */
+  public getWellsByPolygon = async (
+    polygon: string,
+    source: string,
+    crs: string,
+    layerName: string,
+    limit?: number,
+    offset?: number
+  ): Promise<Well[]> => {
+    const geometryBody = {
+      wkt: polygon,
+      crs: crs,
+    };
+
+    const geometryRelBody = {
+      geometry: geometryBody,
+      relation: SpatialRelationshipNameDTO.Within,
+    };
+
+    const points = await geospatialClient.findSpatial({
```

Comment:
Does the JavaScript version of the GeospatialSDK limit the number of of returned points to 1000 like the Python SDK? If it does, do we want to limit the number of wells to 1000? Also, some of the points returned may not correspond to wells, so the number of returned wells may be below the limit even if there are enough in CDF

Reply:
Yea, it is limited by the service contract: [https://docs.cognite.com/api/playground/#operation/searchSpatial](service contract). I added pagination now limited to 1000 pr page